hooks/terraform_checkov.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# globals variables
+# hook ID, see `- id` for details in .pre-commit-hooks.yaml file
+# shellcheck disable=SC2034 # Unused var.
+readonly HOOK_ID='terraform_checkov'
+# shellcheck disable=SC2155 # No way to assign to readonly variable in separate lines
+readonly SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+# shellcheck source=_common.sh
+. "$SCRIPT_DIR/_common.sh"
+
+function main {
+  common::initialize "$SCRIPT_DIR"
+  common::parse_cmdline "$@"
+  # shellcheck disable=SC2153 # False positive
+  common::per_dir_hook "${ARGS[*]}" "$HOOK_ID" "${FILES[@]}"
+}
+
+#######################################################################
+# Unique part of `common::per_dir_hook`. The function is executed in loop
+# on each provided dir path. Run wrapped tool with specified arguments
+# Arguments:
+#   args (string with array) arguments that configure wrapped tool behavior
+#   dir_path (string) PATH to dir relative to git repo root.
+#     Can be used in error logging
+# Outputs:
+#   If failed - print out hook checks status
+#######################################################################
+function per_dir_hook_unique_part {
+  # common logic located in common::per_dir_hook
+  local -r args="$1"
+  # shellcheck disable=SC2034 # Unused var.
+  local -r dir_path="$2"
+
+  # pass git root dir, if args not specified
+  if [ -z "$args" ]; then
+    checkov -d .
+  else
+    # shellcheck disable=SC2068 # hook fails when quoting is used ("$arg[@]")
+    checkov ${args[@]}
+  fi
```

Comment:
Should we be just fine with a simplified conditional like below?
```suggestion
  # shellcheck disable=SC2068 # hook fails when quoting is used ("$arg[@]")
  [[ $@ ]] && checkov ${args[@]} || checkov -d .
```

Reply:
Also how `.` is related to `git root dir`, specifically the `root` bit? (the excerpt is from the comment right before this conditional)

Reply:
Good catch.
Not related at all. I will rewrite that by set default args in `.pre-commit-hooks.yaml`